src/wrfda-ncdiag/wrfda_ncdiag.py

```diff
@@ -0,0 +1,407 @@
+                    tmp[tmp <= wrfda_miss_float] = nc.default_fillvals['f4']
+                else:
+                    tmp = self.df[lvar][:]
+                loc_mdata[loc_mdata_name] = tmp
+
+        # put the TestReference fields in the structure for writing out
+        for tvar in TestVars:
+            test_mdata_name = test_fields[tvar][0]
+            tmp = self.df[tvar][:]
+            tmp[tmp <= wrfda_miss_float] = nc.default_fillvals['f4']
+            test_mdata[test_mdata_name] = tmp
+
+        # check for additional WRFDA output for each variable
+        for wrfdavar, iodavar in wrfda_add_vars.items():
+            if wrfdavar in self.df.variables:
+                tmp = np.transpose(np.asarray(self.df[wrfdavar]))
+
+                tmp[tmp <= wrfda_miss_float] = nc.default_fillvals['f4']
+                for c, chan in enumerate(chanlist):
+                    varname = "brightness_temperature_{:d}".format(chan)
+                    gvname = varname, iodavar
+                    outvals = tmp[c]
+                    outdata[gvname] = outvals
+
+        # tb_obs, tb_err, and tb_qc are nlocs x nchan
+        # --> using transpose speeds up access below
+        obsdata = np.transpose(np.asarray(self.df['tb_obs']))
+        obserr = rad_platform_sensor_ObsError[self.platform_sensor]
+        # obserr  = np.transpose(np.asarray(self.df['tb_err']))
+        obsqc = np.transpose(np.asarray(self.df['tb_qc']))
+
+        # loop through channels for subset
+        var_names = []
+        for c, chan in enumerate(chanlist):
+            value = "brightness_temperature_{:d}".format(chan)
+            var_names.append(value)
+
+            obsdatasub = obsdata[c]
+            obsdatasub[obsdatasub <= wrfda_miss_float] = nc.default_fillvals['f4']
+
+            obserrsub = np.full(self.nlocs, obserr[c])
+            obserrsub[obserrsub <= wrfda_miss_float] = nc.default_fillvals['f4']
+
+            obsqcsub = obsqc[c]
+            obsqcsub[obsqcsub <= wrfda_miss_int] = nc.default_fillvals['i4']
+
+            # store values in output data dictionary
+            outdata[varDict[value]['valKey']] = obsdatasub
+            outdata[varDict[value]['errKey']] = obserrsub
+            outdata[varDict[value]['qcKey']] = obsqcsub.astype(int)
+
+        # var metadata
+        var_mdata['variable_names'] = writer.FillNcVector(var_names, "string")
+        var_mdata['sensor_channel'] = np.asarray(chanlist)
+
+        # dummy record metadata, for now
+        nrecs = 1
+        rec_mdata['rec_id'] = np.asarray([999], dtype='i4')
+        loc_mdata['record_number'] = np.full((self.nlocs), 1, dtype='i4')
+
+        # global attributes
+
+        AttrData["date_time_string"] = self.validtime.strftime("%Y-%m-%dT%H:%M:%SZ")
+        AttrData["satellite"] = self.satellite
+        AttrData["sensor"] = self.sensor
+
+        # set dimension lengths in the writer since we are bypassing
+        # ExtractObsData
+        writer._nrecs = nrecs
```

Comment:
I think we are trying to eliminate the nrecs variable.  See #244.

Reply:
There will need to be a number of changes to the ioda-converters code before nrecs can be eliminated. It's possible that @emilyhcliu already made these changes in order to address #244. In that case we should make a PR from her updates.

Regardless, I think the code changes for eliminating nrecs is outside the scope of this PR and we should do that work in a separate PR (whether it comes from Emily's work or we create a new PR). We can leave in this statement for now, and update accordingly.

Thanks!

Reply:
@srherbener I will submit the PR for #244 today.

Reply:
@emilyhcliu, sounds great. Thanks for taking care of this!